sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/EventDataBatchTests.cs

```diff
@@ -165,6 +180,52 @@ public void TryAddRespectsTheBatchLock()
             Assert.That(batch.TryAdd(new EventData(Array.Empty<byte>())), Is.True, "The event should have been accepted after unlocking.");
         }
 
+        /// <summary>
+        ///   Verifies property accessors for the <see cref="EventDataBatch.Clear" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void ClearRespectsTheBatchLock()
+        {
+            var mockBatch = new MockTransportBatch();
+            var batch = new EventDataBatch(mockBatch, "ns", "eh", new SendEventOptions());
+            var eventData = new EventData(new byte[] { 0x21 });
+
+            Assert.That(batch.TryAdd(new EventData(new byte[] { 0x21 })), Is.True, "The event should have been accepted before locking.");
+
+            batch.Lock();
+            Assert.That(() => batch.Clear(), Throws.InstanceOf<InvalidOperationException>(), "The batch should not accept events when locked.");
+
+            batch.Unlock();
+            batch.Clear();
+
+            Assert.That(mockBatch.ClearInvoked, Is.True, "The batch should have been cleared after unlocking.");
```

Comment:
Not sure if I'm being overconcerned, but maybe for both new batch lock tests we could make sure that the `Invoked` flag is not set after getting the `InvalidOperationException`.

Reply:
That's a good idea.   I'll throw that in.